eksupgrade/models/eks.py

```diff
@@ -0,0 +1,357 @@
+    service_account_role_arn: str = ""
+    publisher: str = ""
+    owner: str = ""
+    product_id: str = ""
+    product_url: str = ""
+    configuration_values: str = ""
+    cluster: Cluster = field(default_factory=lambda: Cluster(arn=""))
+
+    def __repr__(self) -> str:  # pragma: no cover
+        """Return the string representation of a Cluster Addon."""
+        return f"<{self.__class__.__name__} - Name: {self.name} | Version: {self.version} | Last Status: {self.status}>"
+
+    @property
+    def addon_name(self) -> str:
+        """Return the addon name."""
+        return self.name
+
+    @classmethod
+    def get_addon(cls, addon: str, cluster: Cluster, region: str):
+        """Get the cluster addon details and build a ClusterAddon object."""
+        logger.debug("Getting cluster addon details...")
+        _temp_cluster = cls(arn="", region=region)
+        response: DescribeAddonResponseTypeDef = _temp_cluster.eks_client.describe_addon(
+            addonName=addon,
+            clusterName=cluster.name,
+        )
+        addon_data: AddonTypeDef = response["addon"]
+        addon_version: str = addon_data.get("addonVersion", "")
+        markplace_data = addon_data.get("marketplaceInformation", {})
+        logger.debug("Addon: %s - Current Version: %s - Cluster: %s", addon, addon_version, cluster.name)
+
+        return cls(
+            arn=addon_data.get("addonArn", ""),
+            name=addon_data.get("addonName", ""),
+            version=addon_version,
+            status=addon_data.get("status", ""),
+            health=addon_data.get("health", {}),
+            created_at=addon_data.get("createdAt", datetime.datetime.now()),
+            modified_at=addon_data.get("modifiedAt", datetime.datetime.now()),
+            tags=addon_data.get("tags", {}),
+            region=region,
+            service_account_role_arn=addon_data.get("serviceAccountRoleArn", ""),
+            publisher=addon_data.get("publisher", ""),
+            owner=addon_data.get("owner", ""),
+            product_id=markplace_data.get("productId", ""),
+            product_url=markplace_data.get("productUrl", ""),
+            cluster=cluster,
+        )
+
+    @cached_property
+    def _addon_update_kwargs(self) -> Dict[str, Any]:
+        """Get kwargs for subsequent update to addon."""
+        kwargs: Dict[str, Any] = {}
+
+        if self.service_account_role_arn:
+            kwargs["serviceAccountRoleArn"] = self.service_account_role_arn
+        if self.configuration_values:
+            kwargs["configurationValues"] = self.configuration_values
+        return kwargs
+
+    @requires_cluster
+    def update_addon(self, version: str) -> UpdateAddonResponseTypeDef:
+        """Update the addon to the target version."""
+        logger.info("Updating the EKS cluster's %s add-on version via the EKS API...", self.name)
+        update_response: UpdateAddonResponseTypeDef = self.eks_client.update_addon(
```

Comment:
do we know that the version that is being upgraded to is always a valid version for the control plane's Kubernetes version?